src/systems/filecoin_markets/storage_market/storage_market_actor_state.go

```diff
@@ -137,56 +165,74 @@ func (st *StorageMarketActorState_I) _activateDeal(rt Runtime, deal deal.OnChain
 	return deal
 }
 
-func (st *StorageMarketActorState_I) _lockBalance(rt Runtime, addr addr.Address, amount actor.TokenAmount) {
-	if amount < 0 {
-		rt.AbortArgMsg("sma._lockBalance: negative amount.")
-	}
-
-	currBalance := st._safeGetBalance(rt, addr)
+func (st *StorageMarketActorState_I) _rtLockBalance(rt Runtime, addr addr.Address, amount actor.TokenAmount) {
+	Assert(amount >= 0)
+	Assert(st._addressEntryExists(addr))
 
-	if currBalance.Impl().Available() < amount {
+	prevLocked := st._getLockedReqBalance(addr)
+	if prevLocked+amount > st._getTotalEscrowBalance(addr) {
 		rt.AbortFundsMsg("sma._lockBalance: insufficient funds available to lock.")
 	}
 
-	currBalance.Impl().Available_ -= amount
-	currBalance.Impl().Locked_ += amount
+	newLockedReqTable, ok := actor.BalanceTable_WithAdd(st.LockedReqTable(), addr, amount)
+	Assert(ok)
+	st.Impl().LockedReqTable_ = newLockedReqTable
 }
 
-func (st *StorageMarketActorState_I) _unlockBalance(rt Runtime, addr addr.Address, amount actor.TokenAmount) {
-	if amount < 0 {
-		rt.AbortArgMsg("sma._unlockBalance: negative amount.")
-	}
+func (st *StorageMarketActorState_I) _rtUnlockBalance(
+	rt Runtime, addr addr.Address, unlockAmountRequested actor.TokenAmount) {
 
-	currBalance := st._safeGetBalance(rt, addr)
+	Assert(unlockAmountRequested >= 0)
+	Assert(st._addressEntryExists(addr))
 
-	if currBalance.Impl().Locked() < amount {
-		rt.AbortFundsMsg("sma._unlockBalance: insufficient funds to unlock.")
-	}
+	st.Impl().LockedReqTable_ = st._rtTableWithDeductBalanceExact(rt, st.LockedReqTable(), addr, unlockAmountRequested)
```

Comment:
Would be cleaner if the check/failure from lines 211-213 were done here ahead of calling the deduct method.

As in older method of verifying sufficient amount locked is better imo